packages/playwright-core/src/server/registry/index.ts

```diff
@@ -859,6 +851,43 @@ export class Registry {
     };
   }
 
+  async validateHostRequirementsForExecutablesIfNeeded(executables: Executable[], sdkLanguage: string, updateMarkerFile: boolean) {
+    if (getAsBooleanFromENV('PLAYWRIGHT_SKIP_VALIDATE_HOST_REQUIREMENTS')) {
+      process.stderr.write('Skipping host requirements validation logic because `PLAYWRIGHT_SKIP_VALIDATE_HOST_REQUIREMENTS` env variable is set.\n');
+      return;
+    }
+    for (const executable of executables)
+      await this._validateHostRequirementsForExecutableIfNeeded(executable, sdkLanguage, updateMarkerFile);
+  }
+
+  private async _validateHostRequirementsForExecutableIfNeeded(executable: Executable, sdkLanguage: string, updateMarkerFile: boolean) {
+    const kMaximumReValidationPeriod = 30 * 24 * 60 * 60 * 1000; // 30 days
+    // Executable does not require validation.
+    if (!executable.directory)
+      return;
+    const markerFile = path.join(executable.directory, 'DEPENDENCIES_VALIDATED');
+    const depsValidationFileExists: boolean = await fs.promises.access(markerFile).then(() => true).catch(() => false);
+    // Executable is already validated.
+    if (depsValidationFileExists && await fs.promises.stat(markerFile).then(stat => (Date.now() - stat.mtime.getTime()) < kMaximumReValidationPeriod))
+      return;
+
+    debugLogger.log('install', `validating host requirements for "${executable.name}"`);
+    try {
+      await executable._validateHostRequirements(sdkLanguage);
+      debugLogger.log('install', `validation passed for ${executable.name}`);
+    } catch (error) {
+      debugLogger.log('install', `validation failed for ${executable.name}`);
+      throw error;
+    }
+
+    if (updateMarkerFile) {
+      if (depsValidationFileExists)
+        await fs.promises.utimes(markerFile, new Date(), new Date());
+      else
+        await fs.promises.writeFile(markerFile, '');
```

Comment:
Why not just `writeFile` unconditionally?

Reply:
Addressed.